FinalEngine.Editor.ViewModels/MainViewModel.cs

```diff
@@ -19,76 +24,134 @@ namespace FinalEngine.Editor.ViewModels;
 /// <seealso cref="IMainViewModel" />
 public sealed class MainViewModel : ObservableObject, IMainViewModel
 {
+    /// <summary>
+    /// The layout manager factory, used to create an <see cref="ILayoutManager"/> to handle reseting the current window layout.
+    /// </summary>
+    private readonly ILayoutManagerFactory layoutManagerFactory;
+
     /// <summary>
     /// The logger.
     /// </summary>
     private readonly ILogger<MainViewModel> logger;
 
+    /// <summary>
+    /// The view presenter.
+    /// </summary>
+    private readonly IViewPresenter viewPresenter;
+
     /// <summary>
     /// The exit command.
     /// </summary>
     private ICommand? exitCommand;
 
     /// <summary>
-    /// The title of the application.
+    /// The manage window layouts command.
+    /// </summary>
+    private ICommand? manageWindowLayoutsCommand;
+
+    /// <summary>
+    /// The reset window layout command.
+    /// </summary>
+    private ICommand? resetWindowLayoutCommand;
+
+    /// <summary>
+    /// The save window layout command.
     /// </summary>
-    private string? title;
+    private ICommand? saveWindowLayoutCommand;
+
+    /// <summary>
+    /// The toggle tool window command.
+    /// </summary>
+    private ICommand? toggleToolWindowCommand;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="MainViewModel"/> class.
     /// </summary>
     /// <param name="logger">
     /// The logger.
     /// </param>
-    /// <param name="context">
-    /// The application context.
+    /// <param name="viewPresenter">
+    /// The view presenter, used to display views.
+    /// </param>
+    /// <param name="applicationContext">
+    /// The application context, used to get the title of the view.
+    /// </param>
+    /// <param name="layoutManagerFactory">
+    /// The layout manager factory, used to create an <see cref="ILayoutManager"/> to handle reseting the current window layout.
+    /// </param>
+    /// <param name="dockViewModelFactory">
+    /// The dock view model factory used to create an <see cref="IDockViewModel"/> to handle docking of tool and pane views.
     /// </param>
     /// <exception cref="ArgumentNullException">
-    /// The specified <paramref name="logger"/> or <paramref name="context"/> parameter cannot be null.
+    /// The specified <paramref name="logger"/>, <paramref name="viewPresenter"/>, <paramref name="applicationContext"/>, <paramref name="layoutManagerFactory"/> or <paramref name="dockViewModelFactory"/> parameter cannot be null.
     /// </exception>
```

Comment:
💡QUICK TIP💡

You might know this already, but if you want to list items, you can do this.  :)

Great for formatting your docs in your intellisense as well docs generators.

```suggestion
    /// <exception cref="ArgumentNullException">
    /// Occurs when the following arguments are null:
    ///     <list type="bullet">
    ///         <item><paramref name="logger"/></item>
    ///         <item><paramref name="viewPresenter"/></item>
    ///         <item><paramref name="applicationContext"/></item>
    ///         <item><paramref name="layoutManagerFactory"/></item>
    ///         <item><paramref name="dockViewModelFactory"/></item>
    ///     </list>
    /// </exception>
```

This would be the result:
<img width="362" alt="image" src="https://github.com/softwareantics/FinalEngine/assets/85414302/77e43412-1b5c-4091-a41e-855ebe6f7dc5">

Reply:
I didn't know you could do this, I'll create an issue to fix up what classes I currently have where I'm not listing items, thanks!